File: python/questions/maximumHappinessSum.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort(reverse=True)
        cnt = 1
        ans = happiness[0]
        while cnt < k:
            if happiness[cnt] > cnt:
                ans += happiness[cnt] - cnt
                cnt += 1
            else:
                return ans

File: c/questions/maximumHappinessSum.c
#include <stdlib.h>

int camp(const void* a, const void* b) {
    return *(int*)a < *(int*)b;
}

long long maximumHappinessSum(int* happiness, int happinessSize, int k) {
    qsort(happiness, happinessSize, sizeof(int), camp);
    long long ans = happiness[0];
    for (int p=1; p<k; p++) {
        if (happiness[p] > p) {
            ans += happiness[p] - p;
        } else {
            return ans;
        }
    }
    return ans;
}
